src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,229 @@
+
+    val workflowBatchFuture = dataSource.inTransaction { dataAccess =>
+      for {
+      //Load a bunch of things we'll need to reconstruct information:
+      //The list of workflows in this submission
+        wfRecs <- dataAccess.workflowQuery.findWorkflowByIds(workflowIds).result
+        submissionId = wfRecs.head.submissionId
+        _ = assert(wfRecs.forall(_.submissionId == submissionId)) //sanity check they're all in the same submission
+        unpackedWfOpts <- DBIO.sequence(wfRecs.map(dataAccess.workflowQuery.loadWorkflow)) //reify to real workflows
+        _ = assert(unpackedWfOpts.size == wfRecs.size) //there should be the right number of them
+        _ = assert(unpackedWfOpts.forall(_.isDefined)) //and they should all have been found
+        unpackedWfs = for {wfOpt <- unpackedWfOpts; wf <- wfOpt} yield {
+          wf
+        }
+
+        //The submission itself
+        subRecs <- dataAccess.submissionQuery.findById(submissionId).result
+        submissionRec = subRecs.head
+
+        //The workspace
+        wsRecs <- dataAccess.workspaceQuery.findByIdQuery(submissionRec.workspaceId).result
+        workspaceRec = wsRecs.head
+
+        //The workspace's billing project
+        bpOpt <- dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceRec.namespace))
+        billingProject = bpOpt.get
+
+        //The person who submitted the submission, and their token
+        submitterOpt <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(submissionRec.submitterId)))
+        submitter = submitterOpt.get
+        tokenOpt <- DBIO.from(googleServicesDAO.getToken(submitter))
+        token = tokenOpt.get
+
+        //The wdl
+        mcOpt <- dataAccess.methodConfigurationQuery.loadMethodConfigurationById(submissionRec.methodConfigurationId)
+        methodConfig = mcOpt.get
+        wdl <- getWdl(methodConfig)
+      } yield {
+
+        val wfOpts = buildWorkflowOpts(workspaceRec, submissionId, submitter, token, billingProject)
+
+        val wfInputsBatch = unpackedWfs map { wf =>
+          val methodProps = wf.inputResolutions map {
+            case svv: SubmissionValidationValue if svv.value.isDefined =>
+              svv.inputName -> svv.value.get
+          }
+          MethodConfigResolver.propertiesToWdlInputs(methodProps.toMap)
+        }
+
+        //yield the things we're going to submit to Cromwell
+        (wdl, wfRecs.map(_.id), wfInputsBatch, wfOpts)
+      }
+    }
+
+    import ExecutionJsonSupport.ExecutionServiceWorkflowOptionsFormat
+    val cromwellSubmission = for {
+      (wdl: String, workflowRecIds: Seq[Long], wfInputsBatch: Seq[String], wfOpts: ExecutionServiceWorkflowOptions) <- workflowBatchFuture
+      workflowSubmitResult <- executionServiceDAO.submitWorkflows(wdl, wfInputsBatch, Option(wfOpts.toJson.toString), getUserInfo(credential))
+    } yield {
+      workflowRecIds.zip(workflowSubmitResult)
+    }
+
+    cromwellSubmission flatMap { results =>
```

Comment:
in the block of code below you are not chaining your database calls. Each call to dataAccess.XXX does not actually do anything. You need to chain them together with map, flatMap, andThen. As it stands, the only thing that will happen is saving the failures.